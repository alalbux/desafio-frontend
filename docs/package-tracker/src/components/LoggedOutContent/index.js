import React from 'react';
import Button from '@mui/material/Button';
import { useMediaQuery } from '@mui/material';

import LOCALES from '@/locales/navigation';

const { AUTH } = LOCALES;

const LoggedOutContent = ({ setIsLogged }) => {
  const isMobile = useMediaQuery('(max-width:768px)');

  return (
    <div style={isMobile ? styles.mobileContainer : styles.desktopContainer}>
      <Button
        variant="button"
        color="text.primary"
        onClick={() => setIsLogged(true)}
      >
        {AUTH.LOGIN}
      </Button>
      <Button variant="outlined" onClick={() => setIsLogged(false)}>
        {AUTH.SIGNUP}
      </Button>
    </div>
  );
};

const styles = {
  desktopContainer: {
    display: 'flex',
    flex: 2,
    width: 250,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  mobileContainer: {
    display: 'flex',
    flex: 2,
    width: 240,
    alignItems: 'center',
    marginHorizontal: 10,
    paddingHorizontal: 20,
  },
};

export default LoggedOutContent;
