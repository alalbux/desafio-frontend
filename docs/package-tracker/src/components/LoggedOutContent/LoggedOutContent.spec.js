import React from 'react';
import { render } from '@testing-library/react';
import LoggedOutContent from '@/components/LoggedOutContent';

describe('LoggedOutContent', () => {
  it('matches the snapshot', () => {
    const { container } = render(<LoggedOutContent />);
    expect(container).toMatchSnapshot();
  });
});
