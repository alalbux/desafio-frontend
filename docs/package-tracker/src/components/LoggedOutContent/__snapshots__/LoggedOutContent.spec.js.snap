// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedOutContent matches the snapshot 1`] = `
<div>
  <div
    style="display: flex; flex: 2; width: 250px; justify-content: space-between; align-items: center;"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-button MuiButton-buttonText.primary MuiButton-sizeMedium MuiButton-buttonSizeMedium MuiButton-root MuiButton-button MuiButton-buttonText.primary MuiButton-sizeMedium MuiButton-buttonSizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Entrar
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Criar conta
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
